File: compiler/bench/rot5.c
// expected: 200000000
// Rot equivalent on 5 items, 200M times
#include <stdio.h>

int main(void) {
    long a = 0, b = 1, c = 2, d = 3, e = 4;
    long tmp;
    for (long i = 0; i < 200000000L; i++) {
        // rot bottom 3: a b c -> b c a
        tmp = a; a = b; b = c; c = tmp;
        // rot bottom 3: b c a -> c a b
        tmp = a; a = b; b = c; c = tmp;
        // rot bottom 3: c a b -> a b c
        tmp = a; a = b; b = c; c = tmp;
        a = a + 1;
    }
    (void)d;
    (void)e;
    printf("%ld\n", a);
    return 0;
}

File: compiler/bench/rot5.fs
\ expected: 200000000
\ Rot equivalent on 5 items, 200M times
: main
  0 1 2 3 4                 ( a b c d e )
  200000000 0 do
    >r >r rot r> r>         ( b c a d e ) rotate bottom 3
    >r >r rot r> r>         ( c a b d e )
    >r >r rot r> r>         ( a b c d e ) back to original
    >r >r >r >r 1 + r> r> r> r> ( a' b c d e )
  loop
  drop drop drop drop . cr
;
